filter.c

```diff
@@ -26,31 +52,34 @@ int filter(struct xdp_md *ctx)
 	struct iphdr *ip = data + sizeof(struct ethhdr);
 
 	if (ip + 1 > (struct iphdr *)data_end)
-		goto pass;
+		return XDP_DROP;
 
 	if (ip->protocol != IPPROTO_TCP)
-		goto pass;
+		goto accept;
 
 	struct tcphdr *tcp = (void *)ip + (ip->ihl * 4);
 	if (tcp + 1 > (struct tcphdr *)data_end)
-		goto pass;
+		goto accept;
 
 	__u16 dport = bpf_ntohs(tcp->dest);
 	if ((dport != 80 && dport != 443) || !tcp->psh)
-		goto pass;
+		goto accept;
 
 	void *payload = (void *)tcp + (tcp->doff * 4);
 	if (payload > data_end)
-		goto pass;
+		goto accept;
 
 	arg.dport = dport;
 	arg.offset = (__u16)(payload - data);
 	arg.length = (__u16)(data_end - payload);
+	arg.accept = DOME_XDP_ACCEPT;
+	arg.deny = DOME_XDP_DENY;
```

Comment:
I don't think we need this, right?

Reply:
we don't. I just added it for the sake of completion.
will remove then.